### ISSUE TITLE
Docker: 403 error while downloading from S3 / mkdir fail while compiling

### ISSUE THREAD

**Issue by tempico** [nodegit/nodegit#837]
Hello, 
I have dockerfile:

```
FROM ubuntu:latest

RUN apt-get update && apt-get install -y \
  git zip bc curl wget libcurl3 make \
  g++ libssl-dev build-essential \
  python cmake\
  && sync && apt-get clean

RUN set -ex \
  && for key in \
    9554F04D7259F04124DE6B476D5A82AC7E37093B \
    94AE36675C464D64BAFA68DD7434390BDBE9B9C5 \
    0034A06D9D9B0064CE8ADF6BF1747F4AD2306D93 \
    FD3A5288F042B6850C66B31F09FE44734EB7990E \
    71DCFD284A79C3B38668286BC97EC7A07EDE3FC1 \
    DD8F2338BAE7501E3DD5AC78C273792F7D83545D \
  ; do \
    gpg --keyserver ha.pool.sks-keyservers.net --recv-keys "$key"; \
  done

ENV NODE_VERSION 5.2.0
ENV NPM_CONFIG_LOGLEVEL info

RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
  && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
  && gpg --verify SHASUMS256.txt.asc \
  && grep " node-v$NODE_VERSION-linux-x64.tar.gz\$" SHASUMS256.txt.asc | sha256sum -c - \
  && tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
  && rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc

COPY ssh/* /root/.ssh/
COPY init.sh /root/
RUN chmod +x /root/init.sh \
  && mkdir -p /root/.npm \
  && chmod -R 755 /root/.npm \
  && sync \
  && npm i -g grunt-cli nodegit
WORKDIR /usr/src/app/
```

While building docker I have error:
```
[nodegit] Fetching binary from S3.
[nodegit] Failed to install prebuilt binary:
{ [Error: Command failed: /bin/sh -c /usr/local/lib/node_modules/nodegit/node_modules/.bin/node-pre-gyp install --fallback-to-build=false
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.6.17
node-pre-gyp info using node@5.2.0 | linux | x64
node-pre-gyp info check checked for "/usr/local/lib/node_modules/nodegit/build/Release/nodegit.node" (not found)
node-pre-gyp http GET https://nodegit.s3.amazonaws.com/nodegit/nodegit/nodegit-v0.6.2-node-v47-linux-x64.tar.gz
node-pre-gyp http 403 https://nodegit.s3.amazonaws.com/nodegit/nodegit/nodegit-v0.6.2-node-v47-linux-x64.tar.gz
node-pre-gyp ERR! install error
node-pre-gyp ERR! stack Error: 403 status code downloading tarball https://nodegit.s3.amazonaws.com/nodegit/nodegit/nodegit-v0.6.2-node-v47-linux-x64.tar.gz
node-pre-gyp ERR! stack     at Request.<anonymous> (/usr/local/lib/node_modules/nodegit/node_modules/node-pre-gyp/lib/install.js:101:37)
node-pre-gyp ERR! stack     at emitOne (events.js:83:20)
node-pre-gyp ERR! stack     at Request.emit (events.js:170:7)
node-pre-gyp ERR! stack     at Request.onRequestResponse (/usr/local/lib/node_modules/nodegit/node_modules/node-pre-gyp/node_modules/request/request.js:951:10)
node-pre-gyp ERR! stack     at emitOne (events.js:78:13)
node-pre-gyp ERR! stack     at ClientRequest.emit (events.js:170:7)
node-pre-gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:422:21)
node-pre-gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:89:23)
node-pre-gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:312:20)
node-pre-gyp ERR! stack     at emitOne (events.js:78:13)
node-pre-gyp ERR! System Linux 3.19.0-39-generic
node-pre-gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/nodegit/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build=false"
node-pre-gyp ERR! cwd /usr/local/lib/node_modules/nodegit
node-pre-gyp ERR! node -v v5.2.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.17
node-pre-gyp ERR! not ok
]
  killed: false,
  code: 1,
  signal: null,
  cmd: '/bin/sh -c /usr/local/lib/node_modules/nodegit/node_modules/.bin/node-pre-gyp install --fallback-to-build=false' }
[nodegit] Building manually. (You'll be here a while.)
[nodegit] Regenerating and configuring code
npm info it worked if it ends with ok
npm info using npm@3.3.12
npm info using node@v5.2.0
npm ERR! Linux 3.19.0-39-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--ignore-scripts"
npm ERR! node v5.2.0
npm ERR! npm  v3.3.12
npm ERR! path /root/.npm
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall mkdir

npm ERR! Error: EACCES: permission denied, mkdir '/root/.npm'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES: permission denied, mkdir '/root/.npm']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/root/.npm' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /usr/local/lib/node_modules/nodegit/npm-debug.log
```

Is there any way to fix issues? How this is possible, that in docker container (running from root user) there is not enough rights to create a folder?

**Comment by johnhaley81** [comment]
@tempico can you do this with 0.6.3? Looks like 0.6.2 didn't publish binaries correctly.

**Comment by johnhaley81** [comment]
Fix confirmed via #834 

If this turns out to still not work we can reopen and investigate.
